### PR TITLE
Refactor IPAddresses resource

### LINKED ISSUE #2108: Refactor `/api/island/ip-addresses`
# Refactor

## Component(s) to be refactored

- `/api/island/ip-addresses`

## Tasks
- [x] Use DI to register a convention and inject the addresses into the resource's constructor
- [x] Return a list of IPs
- [x] Update the docstring
- [x] Inject the addresses into `LocalMonkeyRunService`'s constructor
    - [x] Add `IPAddresses` to `resources/__init__.py` and shorten all imports
- [x] Address comments and fix UTs (0.5d)

### LINKED ISSUE #2109: Island Version checking
# Refactor

## Component(s) to be refactored

- monkey_island.cc.resources.version_update
- monkey_island.cc.services.version_update

## Tasks
- [x] Add a new object called `Version` with attributes: (0d) @ilija-lazoroski 
  - [x] `version_number` - Passed in to `Version()` constructor
  - [x] `latest_version` - A property that is populated asynchronously upon construction
  - [x] `download_url`- A property that is populated asynchronously upon construction
- [x] Construct `Version()` at Island startup (0d) @ilija-lazoroski 
- [x] Rename VersionUpdate to Version (0d) @ilija-lazoroski 
  - [x] Rename `/api/version-update` to `/api/island/version`
  - [x] Rename `VersionUpdate` to `Version`
  - [x] Rename `resources/version_update.py` to `resources/version.py`
- [x] Use `Version` object in `/api/island/version` endpoint (0d) @ilija-lazoroski 
  - [x] Add `Version` object to DIContainer
  - [x] Rename `newer_version` to `latest_version`
  - [x] Rename `current_version` to `version_number`
  - [x] Modify the UI to compare `version_number` to `latest_version`
  - [x] Add `Version` to `resources/__init__.py` and shorten all imports
- [x] Remove `services/version_update.py` (0d) @ilija-lazoroski 
- [x] Add authentication to `/api/island/version`. There are comments stating that the UI needs to know the version number before authentication, but I don't know why that's the case. (0d)

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes #2108
The last commit is a workaround to get the tests passing for now. There's probably a better way to do it which I don't see right now.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [ ] ~Have you checked that you haven't introduced any duplicate code?~

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running locally
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/monkey_island/cc/resources/root.py [hunk 1]
```diff
@@ -29,7 +32,7 @@ def get(self, action=None):
     @jwt_required
     def get_server_info(self):
         return jsonify(
-            ip_addresses=local_ip_addresses(),
+            ip_addresses=self._local_ips,
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/resources/root.py:35]: If we have a separate endpoint for this then probably we shouldn't include it here

File: monkey/monkey_island/cc/services/run_local_monkey.py [hunk 2]
```diff
@@ -59,7 +65,7 @@ def run_local_monkey(self):
 
         # run the monkey
         try:
-            ip = local_ip_addresses()[0]
+            ip = self._local_ips[0]
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/run_local_monkey.py:68]: Can there be 0 ip's?

File: monkey/monkey_island/cc/resources/root.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/resources/root.py:35]: +1

File: monkey/monkey_island/cc/resources/ip_addresses.py [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
-        Gets the IP addresses of the Island network interfaces
+        Sends the local IP addresses of the Island
 
-        :return: a dictionary with "ip_addresses" key that points to a list of IP's
+        :return: Local IPs
         """
-        local_ips = local_ip_addresses()
 
-        return {"ip_addresses": local_ips}
+        return self._local_ips
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/resources/ip_addresses.py:25]: These IP addresses may not all be local IPs. They may be publicly routable.

File: monkey/monkey_island/cc/services/utils/network_utils.py [hunk 4]
```diff
@@ -58,7 +58,7 @@ def local_ips():
 # lot of times during the report generation. This means that if the interfaces of the Island machine
 # change, the Island process needs to be restarted.
 @lru(maxsize=1)
-def local_ip_addresses() -> Sequence[str]:
+def get_local_ip_addresses() -> Sequence[str]:
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/utils/network_utils.py:61]: These IP addresses may not all be local IPs. They may be publicly routable.

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Fix all references to "local" IPs. They may not all be local.

I fixed the hack in the `StubDIContainer`. See 5f91e39b39420f75b59e42ba8613d6d90457ed0e and a0ea64e6d5adfaa0bca8859fdc1aff53112ce437.

**Comment by VakarisZ** [comment]
@mssalvatore There's still a bug in di_container: line 135 will raise `UnregisteredTypeError`, when in fact we don't know if it's type or convention.

**Comment by mssalvatore** [comment]
> @mssalvatore There's still a bug in di_container: line 135 will raise `UnregisteredTypeError`, when in fact we don't know if it's type or convention.

I'll address it in a PR tomorrow or next week.

**Review by VakarisZ — APPROVED** [review]

File: monkey/monkey_island/cc/services/run_local_monkey.py (hunk 2 above)
Comment (shreyamalviya) [reply monkey/monkey_island/cc/services/run_local_monkey.py:68]: If there are no interfaces with AF_INET addresses... doesn't seem likely.

**Pull request merged**